Add Promise support, bump to 2.0.1

Cretezy opened the pull request:
I've also generated a `yarn.lock` for use with [Yarn](https://yarnpkg.com/). This won't affect anything.

This does add a dependency on `promise`, which adds 0.9KB on the final build size (7.09 KB  ->  7.18 KB).

dchest commented:
Thanks for PR, however this package is targeting really old browsers, and I hate including shims, especially if it requires external dependencies. Since it's simple enough to just wrap it into a promise in user code, I don't think it's necessary.

See also https://github.com/dchest/scrypt-async-js/pull/28

BTW, my newer package in StableLib uses promises exclusively, so you can switch to it instead https://github.com/StableLib/stablelib/tree/master/packages/scrypt

Cretezy commented:
Perfectly understandable. I'm going to maintain my own fork with a more modern toolkit (Rollup, Jest, ES6+)! Thank you for the library.

Cretezy commented:
https://github.com/Cretezy/scrypt-async-modern

Basically I've updated the code to ES6 (still compiled down to ES5) ,moved to Rollup and Jest, and a few cleanups. It's the same code but modernized overall. :)
